accessors/datastream/datastream_accessor_test.go

```diff
@@ -0,0 +1,156 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package datastream_accessor_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"cloud.google.com/go/datastream/apiv1/datastreampb"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/accessors/clients/datastream"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/accessors/clients/operation"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/streaming"
+	datastream_accessor "github.com/GoogleCloudPlatform/spanner-migration-tool/accessors/datastream"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+)
+
+func TestFetchTargetBucketAndPath(t *testing.T) {
+	dstConfig := streaming.DstConnCfg{
+		Location: "region-X",
+		Name: "profile-name",
+		Prefix: "/",
+	}
+	ctx := context.Background()
+	da := datastream_accessor.DatastreamAccessorImpl{}
+	testCases := []struct{
+		name              	string
+		dsm					datastreamclient.DatastreamClientMock
+		connectionProfile   *datastreampb.ConnectionProfile
+		getProfileErr       error
+		expectedBucketName  string
+		expectedPrefix		string	
+		expectError 		bool
+	}{
+		{
+			name: "basic correct",
+			connectionProfile: &datastreampb.ConnectionProfile{Profile: &datastreampb.ConnectionProfile_GcsProfile{GcsProfile: &datastreampb.GcsProfile{Bucket: "bucket", RootPath: "/"}}},
```

Comment:
Please add another test case where rootPath is /test-path, /test_path/, "" (empty string).

Reply:
Whatever is allowed in rootPath basically, it may be possible that rootPath must always start with "/" or something like that in which case we may skip those cases.

Reply:
For concat, a public utils function is used which should be tested independently, adding a all empty string test case though

Reply:
The root path is based on what is returned by the request itself so no special handling is required